apps/css-workshop/src/components/ListItem.astro

```diff
@@ -0,0 +1,69 @@
+---
+type Props = {
+  isActive?: boolean;
+  size?: string;
+  disabled?: boolean;
+  startIconProps?: astroHTML.JSX.HTMLAttributes;
+  endIconProps?: astroHTML.JSX.HTMLAttributes;
+  contentProps?: astroHTML.JSX.HTMLAttributes;
+  actionable?: boolean;
+  description?: string;
+} & astroHTML.JSX.HTMLAttributes;
+
+const {
+  class: className,
+  isActive,
+  size,
+  disabled,
+  startIconProps,
+  endIconProps,
+  actionable,
+  contentProps,
+  description,
+  ...props
+} = Astro.props;
+---
+
+<li
+  class:list={[
+    'iui-list-item',
+    { 'iui-active': isActive },
+    { 'iui-disabled': disabled },
+    { 'iui-large': size === 'large' },
```

Comment:
In a future PR, we should remove all these classes from the CSS and HTML, and instead use data attributes (similar to how it is done in `ListItem.tsx`).

The current version of the CSS supports both classes and data attributes, but that was mainly done for backwards compatibility (which is no longer a concern).

This can be mentioned in the PR description as an "after-PR TODO".

Reply:
I have added this as an after-PR TODO. I apologize for the late response since I have stopped receiving comment notification emails for some reason but I should have checked PRs often.